index.js

```diff
@@ -44,7 +54,33 @@ module.exports = function expressGAuth(options) {
         passportSession(req, res, err => {
           if (err) {
             next(err)
-          } else if (req.user || config.publicEndPoints.includes(req.originalUrl)) {
+          } else if (req.user) {
+            const { tokenExpirationTime, refreshToken } = req.user
+            const now = Date()
+            const isExpired = isAfter(now, tokenExpirationTime)
+            const shouldBeRefreshed = isExpired &&
+              config.googleAuthorizationParams.accessType === 'offline'
+
+            if (shouldBeRefreshed && refreshToken) {
+              refresh.requestNewAccessToken('google', refreshToken,
+              updateAccessTokenCallback(config, req.session, next))
+            } else if (shouldBeRefreshed) {
+              // Access token needs to be updated, but we are missing refresh
+              // token required to do it. Google provides refresh token only
+              // when authorizing the user through consent screen, so we
+              // reauthenticate the user and ask for their consent again in
+              // order to gain refresh token
+              const { googleAuthorizationParams } = config
+              const newGoogleParams = Object.assign(googleAuthorizationParams,
+                { prompt: 'consent' })
+              const newConfig = Object.assign(config,
+                { googleAuthorizationParams: newGoogleParams })
```

Comment:
Why bind `newConfig` at all if the purpose is to just mutate `config`? Can't see `newConfig` being used anywhere.